### PR TITLE
BB-671: Refactoring statistics.js

### PR DESCRIPTION
**Pull request by meziyum**
### Problem [BB-671](https://tickets.metabrainz.org/browse/BB-671)
Across the codebase we have asynchronous code written using promises, which is hard to read and to maintain and error-prone.

All this code needs to be rewritten using the async/await syntax.

### Solution
The route has been refactored to use async/await syntax for better error handing, maintainability and readability. The parts of the code have also been modularized for better readability and maintainability. 

### Areas of Impact
Statistics - [Statistics Route](https://github.com/metabrainz/bookbrainz-site/blob/master/src/server/routes/statistics.js)

### CODE REVIEW

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Great work, thank you !
In particular, 10 extra points for the nice documentation improvements !

One single small bit of feedback:

File: src/server/routes/statistics.js [hunk 1]
```diff
@@ -31,110 +31,143 @@ import target from '../templates/target';
...
-					.where('master', true)
-					.where('bookbrainz.revision.created_at', '>=',
-						utils.getDateBeforeDays(30));
-			})
-				.count().then((Count) =>
-					 ({Count, modelName}))
+/* Get Statistics Page */
+router.get('/', async (req, res, next) => {
+	const {orm} = req.app.locals;
+	const entityModels = commonUtils.getEntityModels(orm);
```
[the remark this answers is not in the document]
Comment (meziyum) [reply src/server/routes/statistics.js]: I have no idea how I missed that one 🥲 
Apologies for the commits being all over the place.
I would make sure for them to be in proper order in the future.

[the remark this answers is not in the document]
Comment (MonkeyDo) [reply src/server/routes/statistics.js]: Don't worry about it too much, that's why we do code reviews :)

As for commits, if the issue is with tests and code linting, you can run commands on your computer to run the tests and make sure the code is formatted appropriately:

- `npm run test` to run the whole test suite
- `npm run test path/to/file/or/folder` to run a specific test file or folder
- `npm run lint` will show you if there are any code formatting issues

**Review by MonkeyDo — APPROVED** [review]
Great work, thank you !

**Pull request merged**